tools/bin/cdk-auto-release/update-changelog.sh

```diff
@@ -0,0 +1,413 @@
+        error_exit "Temporary file is empty after awk processing. Input file ${filepath} might be empty or awk failed silently."
+    fi
+
+    # Replace the original file with the modified temporary file
+    if ! mv "$temp_file" "$filepath"; then
+        # Temp file might still exist if mv fails, but trap should get it on exit
+        error_exit "Failed to move temporary file to overwrite ${filepath}"
+    fi
+
+    # Disable the trap explicitly on successful completion of this function's critical section
+    # to prevent accidental removal if the temp file was already moved.
+    trap - EXIT TERM INT HUP
+
+    log_info "Successfully inserted new row into ${filepath}" >&2
+}
+
+# --- Git Helper Functions ---
+
+# Function to fetch updates from the master branch
+# Usage: fetch_updates
+fetch_updates() {
+    log_info "Fetching updates from origin/master..."
+    if ! git fetch origin master; then
+        error_exit "Failed to fetch from origin master."
+    fi
+}
+
+# Function to find the merge base between HEAD and origin/master
+# Usage: find_merge_base
+# Returns: The merge base commit hash or exits on error
+find_merge_base() {
+    local merge_base
+    log_info "Finding merge base between HEAD and origin/master..."
+    merge_base=$(git merge-base HEAD origin/master)
+
+    if [ -z "$merge_base" ]; then
+      error_exit "Could not find a common ancestor between HEAD and origin/master."
+    fi
+
+    log_info "Merge base commit: $merge_base"
+    echo "$merge_base" # Return merge base commit hash
+}
+
+# Function to find modified connector folders since the merge base
+# Usage: find_modified_connectors <merge_base_commit>
+# Returns: A string containing unique connector folder names (e.g., "source-postgres\ndestination-bigquery")
+find_modified_connectors() {
+    local merge_base_commit="$1"
+    local modified_files connector_folders
+
+    log_info "Checking for modified connector files since merge base ${merge_base_commit}..." >&2
+    modified_files=$(git diff --name-only --diff-filter=M "$merge_base_commit" HEAD -- 'airbyte-integrations/connectors/*') # Be specific
+
+    if [ -z "$modified_files" ]; then
+      log_info "No connector files were modified on the current branch compared to its merge-base with origin/master." >&2
+      echo "" # Return empty string
+    else
+      log_info "Modified connector files found: $modified_files" >&2
+      # Extract unique connector folder names (e.g., source-postgres)
+      connector_folders=$(echo "$modified_files" | \
+          # --- Old Line (remove or comment out) ---
+          # grep -oP 'airbyte-integrations/connectors/\K[^/]+' | \
+          # --- New Line using awk ---
+          awk -F'/' '$1=="airbyte-integrations" && $2=="connectors" && $3!="" { print $3 }' | \
+          sort -u) # Use sort -u for unique sort
+      log_info "Unique modified connector folders: $connector_folders" >&2
+      echo "$connector_folders" # Return the list
+    fi
+}
+
```

Comment:
I think this can be simplified - unless I'm missing something, we know the list of modified connectors from earlier up in the workflow. Can we pass this as an argument into the script instead of recalculating this to simplify everything?

Reply:
Done